source/src/plugins/KolmogorovTest.cc

```diff
@@ -0,0 +1,171 @@
+	              "be used for TGraphs. While this test can take TH1s, the Kolmogorov test is intended for use on unbinned data, not histograms. "
+	              "See https://root.cern.ch/doc/master/classTH1.html#aeadcf087afe6ba203bcde124cfabbee4 for more information.";
+    }
+
+    //-------------------------------------------------------------------------------------------------
+
+    StatusCode KolmogorovTest::readSettings(const TiXmlHandle xmlHandle) {
+      RETURN_RESULT_IF_AND_IF(STATUS_CODE_SUCCESS, STATUS_CODE_NOT_FOUND, !=, XmlHelper::readParameter(xmlHandle, "UseUnderflow", m_useUnderflow));
+      RETURN_RESULT_IF_AND_IF(STATUS_CODE_SUCCESS, STATUS_CODE_NOT_FOUND, !=, XmlHelper::readParameter(xmlHandle, "UseOverflow", m_useOverflow));
+
+      return STATUS_CODE_SUCCESS;
+    }
+
+    std::string KolmogorovTest::getTestOptions(const bool isHistogram, bool useUnderflow, bool useOverflow) {
+      std::string optionsString = "";
+
+      if (useUnderflow and isHistogram) {
+	optionsString += "U";
+      }
+      if (useOverflow and isHistogram) {
+	optionsString += "O";
+      }
+      return optionsString;
+    }
+
+    void KolmogorovTest::userRun(MonitorElement* pMonitorElement, QualityTestReport &report) {
+      const bool hasObject = (pMonitorElement->object() != nullptr);
+      const bool hasReference = (pMonitorElement->reference() != nullptr);
+      const bool isObjHistogram = pMonitorElement->object()->InheritsFrom("TH1");
+      const bool isObjGraph = pMonitorElement->object()->InheritsFrom("TGraph");
+      const bool isRefHistogram = pMonitorElement->reference()->InheritsFrom("TH1");
+      const bool isRefGraph = pMonitorElement->reference()->InheritsFrom("TGraph");
+
+      if (!hasObject) {
+        report.m_message = "No object attached to monitor element";
+        report.m_quality = 0.f;
+        throw StatusCodeException(STATUS_CODE_INVALID_PTR);
+      }
+
+      if (!hasReference) {
+        report.m_message = "No reference attached to monitor element";
+        report.m_quality = 0.f;
+        throw StatusCodeException(STATUS_CODE_INVALID_PTR);
+      }
+
+      if ( (isObjGraph and !isRefGraph) or (isObjHistogram and !isRefHistogram) ) {
+        report.m_message = "The object and reference must be of the same type";
+        report.m_quality = 0.f;
+        throw StatusCodeException(STATUS_CODE_INVALID_PARAMETER);
+      }
+
+      if (isObjGraph) {
+	TGraph* pGraph = pMonitorElement->objectTo<TGraph>();
+	TGraph* pReferenceGraph = pMonitorElement->referenceTo<TGraph>();
+	std::string m_options = KolmogorovTest::getTestOptions(isObjHistogram, m_useUnderflow, m_useOverflow);
+
+	int sizeGraph = pGraph->GetN();
+	int sizeRef = pReferenceGraph->GetN()  ;
+	double* pArrayGraph = pGraph->GetY();
+	double* pArrayRef = pReferenceGraph->GetY();
+
+	std::sort(&pArrayGraph[0], &pArrayGraph[sizeGraph]);
+	std::sort(&pArrayRef[0], &pArrayRef[sizeRef]);
+       
+	report.m_extraInfos = ("Options used: " + m_options);
```

Comment:
m_extraInfos is json object. For this purpose, it can be used as a map : 
```cpp
report.m_extraInfos["options"] = options;
```
See https://github.com/DQM4HEP/dqm4hep-core/blob/master/source/src/plugins/FitParamInRangeTest.cc#L410